homeassistant/components/sensor/statistics.py

```diff
@@ -41,6 +42,8 @@
 DEFAULT_SIZE = 20
 ICON = 'mdi:calculator'
 
+DEPENDENCIES = ['recorder']
```

Comment:
Adding recorder as dependency is a breaking change.

Most people using the statistics sensor probably have it enabled anyway.
If recorder is enabled it is initialized first anyway.

We should either mark the change as breaking or add code that skips `_initzialize_from_database` in case recorder is not enabled.

Reply:
I like the second option better: 
1) How can i find out, if the recorder is enabled?
1) Should the DEPENDENCIES stay as they are? Or do I have to also wrap that in some checks?

Reply:
`if 'recorder' in hass.config.components`

You should remove `recorder` from DEPENDENCIES, otherwise it will force-load it.

Could you test such setup and make sure `recorder` doesn't get loaded?

Reply:
Awesome, Thx :smiley: ! It, works like a charm: 
* If recorder is not configured, it's not loaded. I checked this in the log file.
* If recorder is configured (e.g. via history), the statistics are initialized with the data from the database.

New patch is on the way...